rust/src/lib.rs

```diff
@@ -8,6 +8,18 @@ mod bindings;
 #[repr(transparent)]
 pub struct Plugin(isize);
 
+impl From<isize> for Plugin {
+    fn from(i: isize) -> Self {
+        Plugin(i)
+    }
+}
+
+impl From<Plugin> for isize {
```

Comment:
@nilslice I had to implement From going the other way for `into()` to work. does this seem right?

```
error[E0277]: the trait bound `isize: From<Plugin>` is not satisfied
  --> src/lib.rs:31:33
   |
31 |         Ok(plugin) => Ok(plugin.into())
   |                                 ^^^^ the trait `From<Plugin>` is not implemented for `isize`
   |
   = help: the following other types implement trait `From<T>`:
             <f32 as From<i16>>
             <f32 as From<i8>>
             <f32 as From<u16>>
             <f32 as From<u8>>
             <f64 as From<f32>>
             <f64 as From<i16>>
             <f64 as From<i32>>
             <f64 as From<i8>>
           and 67 others
   = note: required because of the requirements on the impl of `Into<isize>` for `Plugin`
```

Reply:
Yea, sorry if I may have done the reverse in the other PR... You can also use `From::from(plugin)` if only one is available, but I personally tend to prefer using `.into()` when I can as you've done here.